### PR TITLE
Use native Promise instead of when.js

### PR DESCRIPTION
**Pull request by mikach**

### CODE REVIEW

**Comment by mikach** [comment]
@pwagener @cmaher pls review

File: spec/Boss.spec.js [hunk 1]
```diff
@@ -153,7 +152,7 @@ describe('The Boss module', function() {
             });
 
             //noinspection BadExpressionStatementJS
-            expect(when.isPromiseLike(promise)).to.be.true;
+            expect(promise.then).to.be.a('function');
```
Comment (cmaher) [inline spec/Boss.spec.js]: does `expect(promise).to.be.a(Promise)` work?

Comment (mikach) [inline spec/Boss.spec.js]: `expect(promise).to.be.an.instanceof(Promise);` works

**Comment by cmaher** [comment]
I think this is good to merge now

**Comment by pwagener** [comment]
Merging this in now -- thanks!  

I'll publish a new version to NPM shortly after; need to update the docs & the example site to ensure IE 10 can work as well.

**Pull request merged**